src/Model/Release/LibraryRelease.php

```diff
@@ -347,11 +345,13 @@ public function setVersion($version)
     }
 
     /**
-     * True if the current branch ends in the release branch suffix
+     * Check if the current minor branch matches what it expected to be given the new tag in the plan
      */
-    public function isOnReleaseBranch(): bool
+    public function isOnCorrectMinorReleaseBranch(): bool
     {
-        $branch = $this->getLibrary()->getBranch();
-        return preg_match('/' . preg_quote(self::RELEASE_BRANCH_SUFFIX) . '$/', $branch);
+        if (!preg_match('#^([0-9]{1,2}\.[0-9]{1,2})\.[0-9]{1,2}$#', $this->version, $matches)) {
```

Comment:
`$this->version` is what's in the plan, right? Won't this then always return false if we're doing a beta or rc (where version has `-beta1` at the end for example).
If so, is that intentional? I'm trying to understand how this condition is helping us.

Reply:
Ah yep, have updated to remove the trailing '$' so it'll match beta1 etc

Main thing is to extra the major.minor so we get the expected branch name

Reply:
Sweet. I don't think the condition is needed but it also doesn't hurt anything so I'm okay with it staying for this internal thing.